src/Type/Definition/FieldArgument.php

```diff
@@ -82,7 +85,13 @@ public static function createMap(array $config) : array
      */
     public function getType() : Type
     {
-        return $this->type;
+        $type = Schema::resolveType($this->type);
+        assert(
```

Comment:
The assertion stuff is really cool! We could probably utilize it more in future in the lib (as it makes sense to have stricter checks in dev which could be disabled and have 0 cost in prod)

Reply:
The only problem with those is that it persuades phpstan or any other static analysis about something, that is not real in prod runtime. 

Eg. `assert($x !== null)` narrows `int|null` into `int` for phpstan but in reality `null` value can be passed through it.

I still don't know how to approach it. I'm rather using Assertion libraries that throw exception everytime assertion fails currently in my job . But it's not suitable way here either as it would add some cost.

Reply:
You're right, of course. Asserts fall into the same family of precautions as unit tests and static checking: they aren't on the job in a production environment. But in development they're fantastic--they create a strong relation between the static type and the run-time type. 

It's true that there is some room for divergence on production, but remember that these things are designed to catch programming errors, not user errors, so in principle there's only potential for danger here for code that you not only didn't test, but _didn't even try_ in your development environment.

Reply:
I encountered situations where these assertions lied even in tested and dev-run code though 😶

Reply:
Then I suppose we'll have to watch for reports of such cases and adjust our methodology accordingly.